### DISCUSSION TITLE
Possible new features

### DISCUSSION THREAD

**Discussion by jgauchia** [jgauchia/IceNav-v3#82] [Ideas]
* GPX Parsing
* Waypoint save
* Various settings (lcd, compass, sleep...)

**Comment by Shreyashsatone** [comment]
Hello there I am Working with "ELECROW ESP32 Terminal–with ESP32 3.5 inch parallel 480×320 TFT capacitive touch RGB display(RGB by chip ILI9488)" whit bord in the requeriment Display need to use "FT6X36 " con you help gide how to integrate

**Reply by jgauchia** [reply]
Hi!

You can try to use MAKERFABS environment, this board use the same TFT panel and touch.

Please, review hal.cpp for pin assignment.

**Reply by jgauchia** [reply]
Maybe, if you share me info about the board i'll made a branch in order to give support and you can test it

**Reply by jgauchia** [reply]
If you want, please open an issue with all the possible info about your board and I try to work it

**Reply by Shreyashsatone** [reply]
I have Tryed using "MAKERFABS environment" from last Two days I am sucessfull in deploying code but unable to start Desplay screen it is asking for the "FT6X36" library. I have also tryed changing "hal.cpp pin assignment". But it did not workout.

This is the board detile link:-
https://robu.in/product/elecrow-esp32-terminal-with-esp32-3-5-inch-parallel-480x320-tft-capacitive-touch-rgb-displayrgb-by-chip-ili9488/?gad_source=1&gclid=Cj0KCQjwzva1BhD3ARIsADQuPnUBL5Wnh72beYLEnL0SMpd0_4TPQ-3lfNj_WxJGONgNmg5aiVz-tloaAmN6EALw_wcB

**Reply by jgauchia** [reply]
Ok, can you open any issue in GitHub? I'll try to test It

**Reply by Shreyashsatone** [reply]
I Chack for MAKERFABS Board and it in not avable.

**Reply by jgauchia** [reply]
I've a lot of messages about the project, I'm trying to answer all sorry 😟

**Reply by Shreyashsatone** [reply]
I Have add issue.
Thanku for the help

**Reply by jgauchia** [reply]
thank you! 
I'll try to solve it!

**Reply by Shreyashsatone** [reply]
hello,

I have tryed using the code but it did not.

I have use this code "https://github.com/fbiego/windows-10-mobile-lvgl" and it worked on "ELECROW ESP32 Terminal".
Can you help me configr your code so i [IceNav-v3](https://github.com/jgauchia/IceNav-v3) your code on the "ELECROW ESP32 Terminal"

**Reply by jgauchia** [reply]
Hi,

Ok I'll review the link and try to configure it

**Reply by Shreyashsatone** [reply]
Hello sir Can yu tell me how to add [FT6x36](https://github.com/strange-v/FT6X36) driver in the code like-this [driver](https://github.com/jgauchia/IceNav-v3?tab=readme-ov-file#screens) in your code or con you add in the "**devel_support_elecrow_esp32**" Branch.

**Reply by jgauchia** [reply]
The FT6x36 driver is supported by lovyangfx, is the same that FT5x06 .

In environment MAKERF_ESP32S3 if you compile It with correct pin configuration (see hal.cpp) I should be work

**Reply by Shreyashsatone** [reply]
Ok,
I hav tried changin the pin  configuration in hal.cpp

/**
 * @brief TFT 8/16 Bits parallel pin definition
 *
 */
extern const uint8_t TFT_WR  = 18;
extern const uint8_t TFT_RD  = 48;
extern const uint8_t TFT_RS  = 45;
extern const uint8_t TFT_RST = -1;
extern const uint8_t TFT_CS  = -1;

// 8 bit Mode
extern const uint8_t TFT_D0  = 47;
extern const uint8_t TFT_D1  = 21;
extern const uint8_t TFT_D2  = 14;
extern const uint8_t TFT_D3  = 13;
extern const uint8_t TFT_D4  = 12;
extern const uint8_t TFT_D5  = 11;
extern const uint8_t TFT_D6  = 10;
extern const uint8_t TFT_D7  = 9;

// Extra 16 bit Mode
extern const uint8_t TFT_D8  = 3;
extern const uint8_t TFT_D9  = 8;
extern const uint8_t TFT_D10 = 16;
extern const uint8_t TFT_D11 = 15;
extern const uint8_t TFT_D12 = 7;
extern const uint8_t TFT_D13 = 6;
extern const uint8_t TFT_D14 = 5;
extern const uint8_t TFT_D15 = 4;

in this place can you chack and tall if this is correck or not and also tall where eals should i do changes.
🙄

**Reply by jgauchia** [reply]
It's seems to be correct comparated with Windows example link that you write before.

Also review i2c touch pin

**Reply by Shreyashsatone** [reply]
ok I will chack now for  i2c touch pin and update you

**Reply by Shreyashsatone** [reply]
I did not see any  i2c touch pin for ARDUINO_ESP32S3_DEV    // MAKERFABS ESP32S3 too


/**
 * @brief TOUCH I2C pin definition
 *
 */
#ifdef ICENAV_BOARD
  extern const uint8_t TCH_I2C_PORT = 0;
  extern const uint8_t TCH_I2C_SDA  = 38;
  extern const uint8_t TCH_I2C_SCL  = 39;
  extern const uint8_t TCH_I2C_INT  = 40;
#endif
#ifndef ICENAV_BOARD
  extern const uint8_t TCH_I2C_PORT = 0;  
  extern const uint8_t TCH_I2C_SDA  = 38;
  extern const uint8_t TCH_I2C_SCL  = 39;
  extern const uint8_t TCH_I2C_INT  = 40;
#endif

**Reply by jgauchia** [reply]
Under #ifndef ICENAV_BOARD

**Reply by jgauchia** [reply]
Hi!

Have you managed to get it working? I’m really interested in supporting multiple existing boards on the market and making my firmware compatible with them. The problem is that I don't have access to all of them.

That doesn’t help much.

**Reply by Shreyashsatone** [reply]
Hello sir,

I was able to find the log need help to find problem and solution for these

15:15:09.085 -> ESP-ROM:esp32s3-20210327
15:15:09.085 -> Build:Mar 27 2021
15:15:09.085 -> rst:0x1 (POWERON),boot:0xb (SPI_FAST_FLASH_BOOT)
15:15:09.085 -> SPIWP:0xee
15:15:09.085 -> mode:DIO, clock div:1
15:15:09.085 -> load:0x3fce3808,len:0x4bc
15:15:09.085 -> load:0x403c9700,len:0xbd8
15:15:09.085 -> load:0x403cc700,len:0x2a0c
15:15:09.085 -> entry 0x403c98d0
15:15:09.392 -> [   310][V][WiFiServer.h:42] WiFiServer(): WiFiServer::WiFiServer(port=11000, ...)
15:15:09.392 -> E (328) esp_core_dump_flash: Core dump data check failed:
15:15:09.392 -> Calculated checksum='56c8009d'
15:15:09.392 -> Image checksum='ffffffff'
15:15:09.392 -> E (330) psram: PSRAM ID read error: 0x00ffffff, PSRAM chip not found or not supported, or wrong PSRAM line mode
15:15:09.428 -> [   336][W][esp32-hal-psram.c:71] psramInit(): PSRAM init failed!
15:15:09.428 -> [   372][V][esp32-hal-uart.c:330] uartBegin(): UART0 baud(115200) Mode(800001c) rxPin(44) txPin(43)
15:15:09.460 -> [   381][V][esp32-hal-uart.c:416] uartBegin(): UART0 not installed. Starting installation
15:15:09.460 -> [   390][V][esp32-hal-uart.c:463] uartBegin(): UART0 initialization done.
15:15:09.460 -> [   398][I][esp32-hal-i2c.c:75] i2cInit(): Initialising I2C Master: sda=38 scl=39 freq=100000
15:15:09.491 -> [   407][W][Wire.cpp:301] begin(): Bus already started in Master Mode.
15:15:09.491 -> [   425][E][Wire.cpp:499] requestFrom(): i2cWriteReadNonStop returned Error -1
15:15:09.491 -> [   432][E][compass.cpp:75] initCompass(): IMU initialization unsuccessful
15:15:09.491 -> [   438][E][compass.cpp:76] initCompass(): Check IMU wiring or try cycling power
15:15:09.524 -> [   446][E][compass.cpp:77] initCompass(): Status: -1
15:15:09.524 -> [   451][I][EasyPreferences.cpp:28] init(): debug mode:	enable
15:15:09.524 ->

15:15:09.524 -> [   458][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: C_offset_x NOT_FOUND
15:15:09.524 -> [   466][E][Preferences.cpp:503] getBytesLength(): nvs_get_blob len fail: C_offset_y NOT_FOUND
15:15:09.558 -> [   475][V][Preferences.cpp:352] getUChar(): nvs_get_u8 fail: Map_rot NOT_FOUND
15:15:09.558 -> [   482][V][Preferences.cpp:400] getUInt(): nvs_get_u32 fail: Def_zoom NOT_FOUND
15:15:09.558 -> [   489][V][Preferences.cpp:352] getUChar(): nvs_get_u8 fail: Map_compass NOT_FOUND
15:15:09.558 -> [   497][V][Preferences.cpp:352] getUChar(): nvs_get_u8 fail: Compass_rot NOT_FOUND
15:15:09.558 -> [   504][V][Preferences.cpp:352] getUChar(): nvs_get_u8 fail: Map_speed NOT_FOUND
15:15:09.589 -> [   512][V][Preferences.cpp:352] getUChar(): nvs_get_u8 fail: Map_scale NOT_FOUND
15:15:09.589 -> [   519][V][Preferences.cpp:364] getShort(): nvs_get_i16 fail: GPS_speed NOT_FOUND
15:15:09.589 -> [   527][V][Preferences.cpp:364] getShort(): nvs_get_i16 fail: GPS_rate NOT_FOUND
15:15:09.589 -> [   534][V][Preferences.cpp:388] getInt(): nvs_get_i32 fail: Compass_X NOT_FOUND
15:15:09.624 -> [   541][V][Preferences.cpp:388] getInt(): nvs_get_i32 fail: Compass_Y NOT_FOUND
15:15:09.624 -> [   548][V][Preferences.cpp:388] getInt(): nvs_get_i32 fail: Coords_X NOT_FOUND
15:15:09.624 -> [   556][V][Preferences.cpp:388] getInt(): nvs_get_i32 fail: Coords_Y NOT_FOUND
15:15:09.624 -> [   563][V][Preferences.cpp:388] getInt(): nvs_get_i32 fail: Altitude_X NOT_FOUND
15:15:09.624 -> [   570][V][Preferences.cpp:388] getInt(): nvs_get_i32 fail: Altitude_Y NOT_FOUND
15:15:09.657 -> [   577][V][Preferences.cpp:388] getInt(): nvs_get_i32 fail: Speed_X NOT_FOUND
15:15:09.657 -> [   584][V][Preferences.cpp:388] getInt(): nvs_get_i32 fail: Speed_Y NOT_FOUND
15:15:09.657 -> [   591][V][Preferences.cpp:352] getUChar(): nvs_get_u8 fail: Map_vector NOT_FOUND
15:15:09.657 -> [   599][V][Preferences.cpp:352] getUChar(): nvs_get_u8 fail: Map_mode NOT_FOUND
15:15:09.689 -> [   606][V][settings.cpp:281] printSettings():     KEYNAME 	DEFINED 	VALUE
15:15:09.689 -> [   613][V][settings.cpp:282] printSettings():     ======= 	======= 	=====
15:15:09.689 -> [   620][V][settings.cpp:290] printSettings():     Map_rot 	default 	
15:15:09.689 -> [   626][V][settings.cpp:290] printSettings():   Map_speed 	default 	
15:15:09.689 -> [   633][V][settings.cpp:290] printSettings():   Map_scale 	default 	
15:15:09.721 -> [   639][V][settings.cpp:290] printSettings(): Map_compass 	default 	
15:15:09.721 -> [   646][V][settings.cpp:290] printSettings():  Map_vector 	default 	
15:15:09.721 -> [   652][V][settings.cpp:290] printSettings():    Map_mode 	default 	
15:15:09.721 -> [   659][V][settings.cpp:290] printSettings():   Compass_X 	default 	
15:15:09.721 -> [   665][V][settings.cpp:290] printSettings():   Compass_Y 	default 	
15:15:09.753 -> [   672][V][settings.cpp:290] printSettings():    Coords_X 	default 	
15:15:09.753 -> [   678][V][settings.cpp:290] printSettings():    Coords_Y 	default 	
15:15:09.753 -> [   685][V][settings.cpp:290] printSettings():  Altitude_X 	default 	
15:15:09.753 -> [   691][V][settings.cpp:290] printSettings():  Altitude_Y 	default 	
15:15:09.753 -> [   697][V][settings.cpp:290] printSettings():     Speed_X 	default 	
15:15:09.785 -> [   704][V][settings.cpp:290] printSettings():     Speed_Y 	default 	
15:15:09.785 -> [   710][V][settings.cpp:290] printSettings():  C_offset_x 	default 	
15:15:09.785 -> [   717][V][settings.cpp:290] printSettings():  C_offset_y 	default 	
15:15:09.785 -> [   723][V][settings.cpp:290] printSettings(): Compass_rot 	default 	
15:15:09.785 -> [   730][V][settings.cpp:290] printSettings():   GPS_speed 	default 	
15:15:09.823 -> [   736][V][settings.cpp:290] printSettings():    GPS_rate 	default 	
15:15:09.823 -> [   743][V][settings.cpp:290] printSettings():    Def_zoom 	default 	
15:15:09.823 -> [   757][V][storage.cpp:45] initSD(): SD Card Mounted
15:15:09.823 -> [   762][I][storage.cpp:56] initSPIFFS(): Initializing SPIFFS
15:15:09.945 -> [   868][I][storage.cpp:83] initSPIFFS(): Partition size: total: 1438481 used: 502
15:15:10.230 -> [  1174][E][Wire.cpp:137] setPins(): bus already initialized. change pins only when not.
15:15:10.264 -> [  1182][W][Wire.cpp:301] begin(): Bus already started in Master Mode.
15:15:10.264 -> [  1190][V][Touch_FT5x06.cpp:63] _check_init(): [FT5x06] CIPHER:0x64 / FIRMID:0x01 / VENDID:0x11
15:15:10.336 -> [  1249][E][tft.cpp:87] touchCalibrate(): Touch calibration doesn't exists

**Comment by jgauchia** [comment]
This log is normal if you don't have any preferences saved in NVS. in other words, independent of this logs , do you have any image on the TFT?

**Reply by Shreyashsatone** [reply]
No, i can't able to see any image on TFT

**Reply by jgauchia** [reply]
ok, , I'll try to work on it,

**Reply by Shreyashsatone** [reply]
thank you vary mush I will weith, I will also try to figer out. 
also I have a Q. can it be because of the PSRAM MAKERFABS have 2M Elecrow have 8M.

**Reply by jgauchia** [reply]
No problem, code search for available PSRAM and adapt it, but it's possible that environment fails.

I'll check it

**Reply by jgauchia** [reply]
Please, can you send me an email? I'll Talk to you about your board

**Reply by Shreyashsatone** [reply]
please go through this "https://github.com/fbiego/windows-10-mobile-lvgl" I tried to copy environment from this and running icenav lvgl code but still failed. See if you can get it working.

**Reply by Shreyashsatone** [reply]
email Id:- kaisatone143@gmail.com

**Comment by jgauchia** [comment]
Hi,

Friday I'll receive an ELECROW ESP32 Terminal–with ESP32 3.5 board, when arrives I'll work on it to bring it support.

Please, be patient..

Thank you

**Comment by jgauchia** [comment]
Hi, 

Today I've receive ELECROW board, this afternoon I'll try to modify my firmware I'll tell you when I'll finish
